pep-0622.rst

```diff
@@ -21,103 +21,254 @@ Resolution:
+and one or more ``case`` clauses, where each case clause specifies a
+pattern (the overall shape to be matched), an optional “guard” (a
+condition to be checked if the pattern matches), and a code block to
+be executed if the case clause is selected.
+
+The rest of the PEP motivates why we believe pattern matching makes a
+good addition to Python, explains our design choices, and contains a
+precise syntactic and runtime specification. We also give guidance for
+static type checkers (and one small addition to the ``typing`` module)
+and discuss the main objections and alternatives that have been
+brought up during extensive discussion of the proposal, both within
+the group of authors and in the python-dev community. Finally, we
+discuss some possible extensions that might be considered in the
+future, once the community has ample experience with the currently
+proposed syntax and semantics.
+
+
+Overview
+========
 
-:pep:`275` and :pep:`3103` previously proposed similar constructs, and
-were rejected.  Instead of targeting the optimization of
-``if ... elif ... else`` statements (as those PEPs did), this design
-focuses on generalizing sequence, mapping, and object destructuring.
-It uses syntactic features made possible by :pep:`617`, which
-introduced a more powerful method of parsing Python source code.
+Since patterns are a new syntactical category, with their own rules
+and exceptions, and since they mix input (given values) and output
+(captured variables) in novel ways, they require a bit of getting used
+to. It is the experience of the authors that this happens quickly when
+a brief introduction to the basic concepts such as the following is
+presented. Note that this section is not intended to be complete or
+perfectly accurate.
+
+A new syntactic construct called *pattern* is
+introduced. Syntactically patterns look like a subset of expressions;
+the following are patterns:
+
+- ``[first, second, *rest]``
+- ``Point2d(x, 0)``
+- ``{"name": "Bruce", "age": age}``
+- ``42``
+
+The above look like examples of object construction. A constructor
+takes some values as parameters and builds an object from those
+components. But as a pattern the above mean the inverse operation of
+construction which we call *destructuring*: it takes a subject object
+and extracts its components. The syntactic similarity between
+construction and destructuring is intentional and follows the existing
+Pythonic style which makes assignment targets (write contexts) look
+like expressions (read contexts). Pattern matching never creates
+objects, in the same way that ``[a, b] = my_list`` doesn't create a
+new ``[a, b]`` list, nor reads the values of ``a`` and ``b``.
+
+The intuition we are trying to build in users as they learn this is
+that matching a pattern to a subject binds the free variables (if any)
+to subject components in a way that would result in the original
+subject if the pattern was read as an expression. During this process,
```

Comment:
I don't want to confuse people into thinking they're *always* equivalent. This is clearer to me:

```suggestion
The intuition we are trying to build in users as they learn this is
that matching a pattern to a subject binds the free variables (if any)
to subject components in a way that reflects the original
subject when read as an expression. During this process,
```